internal/magic/archive.go

```diff
@@ -41,6 +41,8 @@ var (
 	Cpio = prefix([]byte("070707"), []byte("070701"), []byte("070702"))
 	// RAR matches a RAR archive file.
 	RAR = prefix([]byte("Rar!\x1A\x07\x00"), []byte("Rar!\x1A\x07\x01\x00"))
+	// InstallShieldCab matches an InstallShield Cabinet archive file.
+	InstallShieldCab = prefix([]byte("ISc("))
```

Comment:
A text file containing just the string "ISc(" is detected as InstallShieldCab too and it shouldn't.

To fix this, please add a check for InstallShield version like [this](https://github.com/file/file/blob/25d5933179067a438c812637e230353e1bdae38d/magic/Magdir/archive#L828-L830). 
The bytes from 4th to 8th in input, interpreted as little endian integer, should have one of the nine values: 0x1005201, 0x100600c, 0x1007000, 0x1009500, 0x2000578, 0x20005dc, 0x2000640, 0x40007d0, 0x4000834.

Reply:
I believe it's not an exhaustive list of possible versions, more over, looks like these versions are not real. I reviewed dozen cab files and saw only 04 00 00 01 and 01 52 00 01 positions 4 to 7 what corresponds to versions 4.0 and 1.52 accordingly. So, looks like version specified in two bytes and next two bytes are always(?) 00 01.

Reply:
> A text file containing just the string "ISc(" is detected as InstallShieldCab too and it shouldn't.
> 
> To fix this, please add a check for InstallShield version like [this](https://github.com/file/file/blob/25d5933179067a438c812637e230353e1bdae38d/magic/Magdir/archive#L828-L830). The bytes from 4th to 8th in input, interpreted as little endian integer, should have one of the nine values: 0x1005201, 0x100600c, 0x1007000, 0x1009500, 0x2000578, 0x20005dc, 0x2000640, 0x40007d0, 0x4000834.

I believe it's not an exhaustive list of possible versions, more over, looks like these versions are not real. I reviewed dozen cab files and saw only 04 00 00 01 and 01 52 00 01 positions 4 to 7 what corresponds to versions 4.0 and 1.52 accordingly. So, looks like version specified in two bytes and next two bytes are always(?) 00 01.

Reply:
@gabriel-vasile I believe it's not an exhaustive list of possible versions, more over, looks like these versions are not real. I reviewed dozen cab files and saw only 04 00 00 01 and 01 52 00 01 positions 4 to 7 what corresponds to versions 4.0 and 1.52 accordingly. So, looks like version specified in two bytes and next two bytes are always(?) 00 01.

Reply:
@ysyrota the list of version I previously mentioned is, indeed, incomplete.
But I did some digging and this is what I found:

There is no official documentation for this file format. The only documentation is the source code for some tools that reverse-engineered the format.
The original cab tools, called i5comp and i6comp, dated from 1999 and 2003 both have a check like the one you suggested: byte 6 is 0x00 and byte 7 is 0x01.
You can find the source code for these tools on the internet. The check is in src/I5COMP.C and src/i6comp.c in ReadCabHeader function.

But this check fails for newer cab versions.
I installed the trial for Installshield 2021 and created a few cabs. These files had bytes `8c 0a 00 04` at index 4 to 7. [data1.zip](https://github.com/gabriel-vasile/mimetype/files/8549582/data1.zip) is an example.

[github.com/twogood/unshield](https://github.com/twogood/unshield/) seems to be up-to-date with latest InstallShield versions. 
Here is [their code for handling version bytes](https://github.com/twogood/unshield/blob/4852238d1d9aa736b48f0e7405c19ca6a0996b35/lib/libunshield.c#L273-L287) and these are some issues where they discussed adding support for newer versions of InstallShield:
https://sourceforge.net/p/synce/patches/19/
https://sourceforge.net/p/synce/bugs/204/
https://sourceforge.net/p/synce/patches/30/

To summarize, a check we can add to stop plain-text files from being detected as cab is to verify byte at index 7 has value: 0x01, 0x02 or 0x04.

Let me know what you think.

Reply:
That's a thorough research.

I've added check for bytes and indexes 6 an 7 to be safe. Give me know if you think it is too much and we should check only 7th position.